src/clj/rems/db/api_key.clj

```diff
@@ -1,10 +1,67 @@
 (ns rems.db.api-key
-  (:require [rems.db.core :as db]
-            [rems.json :as json]))
+  (:require [clojure.test :refer :all]
+            [rems.db.core :as db]
+            [rems.json :as json]
+            [rems.util :refer [update-present]]))
 
-(defn valid? [key]
-  (not (nil? (db/get-api-key {:apikey key}))))
+(defn- format-api-key [key]
+  (-> key
+      (update-present :users json/parse-string)
+      (update-present :paths json/parse-string)))
 
-(defn add-api-key! [key comment]
+(defn get-api-key [key]
+  (format-api-key (db/get-api-key {:apikey key})))
+
+(defn get-api-keys []
+  (mapv format-api-key (db/get-api-keys {})))
+
+(defn- method= [method pattern]
+  (let [normalized (.toLowerCase (name method))
+        normalized-pattern (.toLowerCase (name pattern))]
+    (or (= normalized-pattern "any")
+        (= normalized-pattern normalized))))
+
+(defn- allowed-by [method path pattern]
+  (and
+   (method= method (:method pattern))
+   (re-matches (re-pattern (:path pattern)) path)))
+
+(deftest test-allowed-by
+  (testing "simple pattern"
+    (is (allowed-by :get "/foo" {:method "get" :path "/foo"}))
+    (is (allowed-by "GET" "/foo" {:method "get" :path "/foo"}))
+    (is (not (allowed-by :get "/foo/" {:method "get" :path "/foo"}))) ;; NB!
```

Comment:
I would document the NB in the test description. Also it does not really answer why this or that.

Reply:
Or maybe it's own `testing` case with the description.

Reply:
done